### PR TITLE
bitnami/thanos enable custom secretName for query.ingress and query.ingress.grpc

### PR DESCRIPTION
**Pull request by ChristianBieri1995**
### Description of the change

The suggested change allows the definition of custom secretNames for query.ingress and query.ingress.grpc while keeping backward compatibility.

### Benefits

* Possibility to use custom secret name(s) instead of simply query.ingress.hostname-tls / query.ingress.grpc.hostname-tls
* Backward compatible since default values are set and left unchanged compared to old naming concept

### Possible drawbacks

* n/a

### Applicable issues

* n/a

### Additional information

* n/a

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/thanos/values.yaml [hunk 1]
```diff
@@ -657,6 +657,9 @@ query:
     ## @param query.ingress.hostname Default host for the ingress resource
     ##
     hostname: thanos.local
+    ## @param query.ingress.secretName Custom secretName for the ingress resource
+    ## If query.ingress.secretName ist not set, the secret will be named as follows: query.ingress.hostname-tls
```
Comment (FraPazGal) [inline bitnami/thanos/values.yaml:661]:
```suggestion
    ## If query.ingress.secretName is not set, the secret will be named as follows: query.ingress.hostname-tls
```

File: bitnami/thanos/values.yaml [hunk 2]
```diff
@@ -756,6 +759,9 @@ query:
       ## @param query.ingress.grpc.hostname Default host for the ingress resource (GRPC)
       ##
       hostname: thanos-grpc.local
+      ## @param query.ingress.grpc.secretName Custom secretName for the ingress resource (GRPC)
+      ## If query.ingress.grpc.secretName ist not set, the secret will be named as follows: query.ingress.grpc.hostname-tls
```
Comment (FraPazGal) [inline bitnami/thanos/values.yaml:763]:
```suggestion
      ## If query.ingress.grpc.secretName is not set, the secret will be named as follows: query.ingress.grpc.hostname-tls
```

**Review by FraPazGal — COMMENTED** [review]
Thanks for the contribution @ChristianBieri1995! I added a few comments to the review, please check them when you can.

File: bitnami/thanos/values.yaml (hunk 1 above)
Comment (ChristianBieri1995) [reply bitnami/thanos/values.yaml:661]: embarrasing typo #1

File: bitnami/thanos/values.yaml (hunk 2 above)
Comment (ChristianBieri1995) [reply bitnami/thanos/values.yaml:763]: embarrasing typo #2

File: bitnami/thanos/Chart.yaml [hunk 3]
```diff
@@ -30,4 +30,4 @@ maintainers:
 name: thanos
 sources:
   - https://github.com/bitnami/charts/tree/main/bitnami/thanos
-version: 12.11.0
+version: 12.11.1
```
[the remark this answers is not in the document]
Comment (ChristianBieri1995) [reply bitnami/thanos/Chart.yaml:33]: (Y)

File: bitnami/thanos/templates/query/ingress-grpc.yaml [hunk 4]
```diff
@@ -54,7 +54,7 @@ spec:
     {{- if and .Values.query.ingress.grpc.tls (or (include "common.ingress.certManagerRequest" ( dict "annotations" .Values.query.ingress.grpc.annotations )) .Values.query.ingress.grpc.selfSigned) }}
     - hosts:
         - {{ .Values.query.ingress.grpc.hostname }}
-      secretName: {{ printf "%s-tls" .Values.query.ingress.grpc.hostname }}
+      secretName: {{ .Values.query.ingress.grpc.secretName | default (printf "%s-tls" .Values.query.ingress.grpc.hostname) }}
```
[the remark this answers is not in the document]
Comment (ChristianBieri1995) [reply bitnami/thanos/templates/query/ingress-grpc.yaml:57]: I'll check that ASAP :)

File: bitnami/thanos/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChristianBieri1995) [reply bitnami/thanos/Chart.yaml]: Done

File: bitnami/thanos/templates/query/ingress-grpc.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ChristianBieri1995) [reply bitnami/thanos/templates/query/ingress-grpc.yaml:57]: Adjusted bitnami/thanos/templates/query/tls-secrets-grpc.yaml and bitnami/thanos/templates/query/tls-secrets.yaml to only be created when query.ingress.grcp.secretName and query.ingress.secretName is set. Hope it's fine this way.

**Pull request merged**